packages/compass-e2e-tests/tests/collection-import.test.ts

```diff
@@ -35,39 +112,186 @@ describe('Collection import', function () {
+    await insertDialog.waitForDisplayed();
+
+    // set the text in the editor
+    await browser.setAceValue(Selectors.InsertJSONEditor, json);
+
+    // make sure that there's an error and that the insert button is disabled
+    const errorElement = await browser.$(Selectors.InsertDialogErrorMessage);
+    await errorElement.waitForDisplayed();
+    expect(await errorElement.getText()).to.equal(
+      'Insert not permitted while document contains errors.'
+    );
+    const insertButton = await browser.$(Selectors.InsertConfirm);
+    await insertButton.waitForEnabled({ reverse: true });
+
+    // cancel and wait for the modal to go away
+    await browser.clickVisible(Selectors.InsertCancel);
+    await insertDialog.waitForDisplayed({ reverse: true });
   });
 
   it('supports JSON files', async function () {
     const jsonPath = path.resolve(__dirname, '..', 'fixtures', 'listings.json');
 
     await browser.navigateToCollectionTab('test', 'json-file', 'Documents');
 
+    await importJSONFile(browser, jsonPath);
+
+    const messageElement = await browser.$(
+      Selectors.DocumentListActionBarMessage
+    );
+    const text = await messageElement.getText();
+    expect(text).to.equal('Displaying documents 1 - 20 of 16116');
+
+    const result = await getFirstListDocument(browser);
+
+    expect(result._id).to.exist;
+    delete result._id;
+
+    expect(result).to.deep.equal({
+      id: '2818',
+      name: '"Quiet Garden View Room & Super Fast WiFi"',
+      host_id: '3159',
+      host_name: '"Daniel"',
+      neighbourhood: '"Oostelijk Havengebied - Indische Buurt"',
+      neighbourhood_group: '""',
+      latitude: '52.36435',
+      longitude: '4.94358',
+      room_type: '"Private room"',
+      price: '59',
+      minimum_nights: '3',
+      number_of_reviews: '280',
+      last_review: '"2019-11-21"', // NOTE: wasn't automatically interpreted as a date
+      reviews_per_month: '2.86',
+      calculated_host_listings_count: '1',
+      availability_365: '124',
+      number_of_reviews_ltm: '2',
+      license: '"0363 5F3A 5684 6750 D14D"',
+    });
+  });
+
+  it('supports JSON files with extended json', async function () {
+    const jsonPath = path.resolve(
+      __dirname,
+      '..',
+      'fixtures',
+      'extended-json.json'
+    );
+
+    await browser.navigateToCollectionTab(
+      'test',
+      'extended-json-file',
+      'Documents'
+    );
+
+    await importJSONFile(browser, jsonPath);
+
+    const messageElement = await browser.$(
+      Selectors.DocumentListActionBarMessage
+    );
+    const text = await messageElement.getText();
+    expect(text).to.equal('Displaying documents 1 - 1 of 1');
+
+    const result = await getFirstListDocument(browser);
+
+    expect(result._id).to.exist;
+    delete result._id;
+
+    // Array and Object could be expanded, but this is probably good enough.
+    // maxKeyField and minKeyField both display as `t()` for some reason.
```

Comment:
Is there a ticket for this? It feels like a pretty clear-cut bug, but I couldn’t find one :)

Reply:
https://jira.mongodb.org/browse/COMPASS-5514